pyproject.toml

```
@@ -38,7 +38,7 @@ known_third_party = ["diastatic-malt", "jax", "jaxlib", "numpy", "pennylane"]
 skips = ["B607"]
 
 [build-system]
-requires = ["setuptools>=62", "wheel", "pybind11>=2.7.0", "numpy>=1.22,<2"]
+requires = ["setuptools>=62", "wheel", "pybind11>=2.7.0", "numpy>=1.22,<=2"]
```

```suggestion
requires = ["setuptools>=62", "wheel", "pybind11>=2.7.0", "numpy==2"]
```

I think this makes more sense? Because numpy is backward compatible. So ideally we want to compile with numpy==2 since we use numpy's C-API.